### PR TITLE
Unify test_data folder across jbrowse-desktop and jbrowse-web

### PR DESCRIPTION
**Pull request by cmdcolin**
This unifies test_data across jbrowse-desktop and jbrowse-web

It also makes it so volvox data is only loaded in development

Note that adding NODE_ENV=development is needed to make electron.js see this flag since it might not see the webpack'd version of that env variable.

### CODE REVIEW

**Comment by cmdcolin** [comment]
I may do a bit of work to make the base config more stripped down with less example data in the human, so that we can have a safe human instance

File: packages/jbrowse-desktop/package.json [hunk 1]
```diff
@@ -53,7 +54,7 @@
   "scripts": {
     "start": "concurrently \"yarn serve\" \"wait-on http://localhost:3000 && yarn develop\"",
     "serve": "cross-env BROWSER=none yarn cra-start",
-    "develop": "electron .",
+    "develop": "NODE_ENV=development ELECTRON_ENABLE_LOGGING=1 electron .",
```
[the remark this answers is not in the document]
Comment (cmdcolin) [reply packages/jbrowse-desktop/package.json]: Logging from electron.js did not show up without this. I find myself often in a case where I am just sort of baffled by not seeing messages. I can make ELECTRON_ENABLE_LOGGING=1 in my env if needed

[the remark this answers is not in the document]
Comment (garrettjstevens) [reply packages/jbrowse-desktop/package.json]: That's odd, any `console.log`s in electron.js show up in the console without enabling that for me:

image

[the remark this answers is not in the document]
Comment (cmdcolin) [reply packages/jbrowse-desktop/package.json]: ok, i guess it works...i reverted

**Comment by cmdcolin** [comment]
I added some code here to address #644 which allows connections to be used across refreshes. Before this, any track that is opened from a connection will produce errors after refreshing the session.

I am not sure if it is the "final word" in this issue but it works well enough to demo opening a track from a connection, and reloading. I have tested BigWig, BigBed, and BAM. It does this by generating specific reproducible configId for the tracks, and that alleviates some of the severe errors. 

It also has to, for the BasicTrack, manually instantiate it's renderer configId because that also caused issue

**Comment by cmdcolin** [comment]
Note:

This issue includes #643 as it was helpful for testing

**Comment by cmdcolin** [comment]
This now has unified test_data across jbrowse-desktop and jbrowse-web

I think it is a valuable thing to keep things from going stale. Note that references to localPath used in jbrowse-desktops config were changed to uri's again. This is with the assumption that the volvox sample data will be removed in prod, so it won't have to evaluate local uri paths.

**Comment by rbuels** [comment]
There's a lot more in this PR than just test_data work, what all is in here?

**Comment by cmdcolin** [comment]
It includes #643 PR

I thought I could elucidate what "this PR only is" by diffing against it but it still is not a clean diff
(https://github.com/GMOD/jbrowse-components/compare/636_trackhubregistry_use_openLocation...unify_test_data)

Still there are a bunch of rider fixes. Here is a shortlist


a) object-hashing configs which fixes #644 
b) couple random cleanups e.g. changes to alignmentmodel. If interested I can prune this PR to
c) base feature detail drawer modification for some sample data I saw
d) made fontWidth scale factor 0.55->0.6 because I saw features overlapping, this changed a couple tests
e) added some human test tracks including MANE annotation, GAPS track
f) made NCListAdapter feature ID include the configId in them because I saw clear unrelated features  across tracks being selected on feature click (e.g. easy id collisions)
g) made it so that features without CDS are given segments glyph instead of just a bare line by searching if any CDS subfeature exists. Makes non-coding feature rendering much better
h) made it so the hg19 reference uses 1 instead of chr1 for the reference names. this enables a COSMIC jbrowse1 datahub connection to be made. also our grch38 already used just 1 instead of chr1 so it is more consistent this way

**Comment by cmdcolin** [comment]
Also as far as the actual unifying, I incorporated also the removal of the volvox data in production

This is done by modularizing out a config_in_dev.json that is merged into config.json depending on ifDev is true

The merging is a custom merge function that just appends data from matching assembly names in the datasets of the config, plus savedSessions. It's possible if the datasets section of the config was a map keyed on assembly name, the merge could be more automatic.

**Pull request merged**